### PR TITLE
Improve RedisVL filter handling, docs, and examples

### PR DESCRIPTION
**Pull request by tylerhutcherson**
Addresses https://github.com/RedisVentures/redisvl/issues/71 which highlights the need to support more edge case queries like "empty queries" or more sophisticated text LIKE query patterns.

- Improves module readability and testability for the individual filter classes.
- Adds support for various filter edge cases (described in the issue above).
- Also fixes a bug in the index creation routine that originally ignored the `overwrite=False` case.
- Adds more docstrings, unit tests and integration tests to improve our coverage on these cases.
- Improves documentation coverage for hybrid queries and getting started.

### CODE REVIEW

File: redisvl/index.py [hunk 1]
```diff
@@ -510,14 +520,26 @@ async def create(self, overwrite: Optional[bool] = False):
         Raises:
             redis.exceptions.ResponseError: If the index already exists.
         """
-        exists = await self.exists()
-        if exists and overwrite:
+        # TODO - enable async version of this
+        # check_redis_modules_exist(self._redis_conn)
+
+        if not self._fields:
```
Comment (Spartee) [inline redisvl/index.py:526]: should this be allowed?

File: redisvl/index.py [hunk 2]
```diff
@@ -499,6 +508,7 @@ def connect(self, url: Optional[str] = None, **kwargs):
             ValueError: If no Redis URL is provided and REDIS_URL env var is not set.
         """
         self._redis_conn = get_async_redis_connection(url, **kwargs)
+        return self
```
Comment (Spartee) [inline redisvl/index.py:511]: why?

File: redisvl/index.py [hunk 3]
```diff
@@ -317,6 +318,7 @@ def connect(self, url: Optional[str] = None, **kwargs):
             ValueError: If the REDIS_URL env var is not set and url is not provided.
         """
         self._redis_conn = get_redis_connection(url, **kwargs)
+        return self
```
Comment (Spartee) [inline redisvl/index.py:321]: this feels odd given that then... it's a classmethod ... kindaof

Is this to make things more fluent?

**Review by Spartee — DISMISSED** [review]
I'm approving, however, I think we should leave out the index updating (or maybe move somewhere other than getting started). I think it makes things seem more complicated and I think we want that to be the lowest bar to cross possible.

Other than that, I pushed up a few changes. Biggest thing is the change of ``_set_value``. Essentially, I don't think it's necessary to treat anything except for tag differently. I've made a few tests for this as well and updated the hybrid query notebook.

File: redisvl/index.py (hunk 3 above)
Comment (tylerhutcherson) [reply redisvl/index.py:321]: yes, it reduces the potential lines of code for a user by allowing `index = SearchIndex.from_dict(...).connect(...)`

File: redisvl/index.py (hunk 2 above)
Comment (tylerhutcherson) [reply redisvl/index.py:511]: See above

File: redisvl/index.py (hunk 1 above)
Comment (tylerhutcherson) [reply redisvl/index.py:526]: This is the current behavior of the synchro version of the `create` method, and was just missing here

**Review by Spartee — APPROVED** [review]
LGTM
